runbooks/sectigo-root-ca-rotation-2025/linux.sh

```diff
@@ -0,0 +1,181 @@
+#!/bin/bash
+# Datadog Agent v5 runbook — checks only logs since this script's restart.
+# Hardened: uses sudo where needed, avoids assoc arrays, robust log slicing.
+
+TARGET_DIR="/opt/datadog-agent/agent"
+TARGET_FILE="${TARGET_DIR}/datadog-cert.pem"
+URL="https://raw.githubusercontent.com/DataDog/dd-agent/master/datadog-cert.pem"
+CONF_FILE="/etc/dd-agent/datadog.conf"
+LOG_FILES="/var/log/datadog/forwarder.log /var/log/datadog/collector.log"
+
+TEMP_INSTALLED=0
+DOWNLOADER=""
+OFFSETS_FILE="$(mktemp -t dd_offsets_XXXXXX)"
+PRE_TS_UNIX=""
+PRE_TS_READABLE=""
+REMOVE_CMD=""
+
+error_exit() {
+  echo "$1" >&2
+  echo "Please contact support for further help." >&2
+  exit 1
+}
+
+install_curl() {
+  echo "Neither curl nor wget found. Attempting to install curl temporarily..."
+  if command -v apt-get &>/dev/null; then
+    INSTALL_CMD="sudo apt-get update && sudo apt-get install -y curl"
+    REMOVE_CMD="sudo apt-get remove -y curl"
+  elif command -v yum &>/dev/null; then
+    INSTALL_CMD="sudo yum install -y curl"
+    REMOVE_CMD="sudo yum remove -y curl"
+  elif command -v dnf &>/dev/null; then
+    INSTALL_CMD="sudo dnf install -y curl"
+    REMOVE_CMD="sudo dnf remove -y curl"
+  else
+    error_exit "Error: No supported package manager found to install curl. Please install curl or wget manually."
+  fi
+  if eval "$INSTALL_CMD"; then
+    echo "Successfully installed curl."
+    TEMP_INSTALLED=1
+    DOWNLOADER="curl"
+  else
+    error_exit "Error: Failed to install curl."
+  fi
+}
+
+check_downloader() {
+  if command -v curl &>/dev/null; then
+    DOWNLOADER="curl"
+  elif command -v wget &>/dev/null; then
+    DOWNLOADER="wget"
+  else
+    install_curl
+  fi
+}
+
+ensure_target_directory() {
+  if ! sudo test -d "$TARGET_DIR"; then
```

Comment:
the command is being run with `sudo`, I'm not sure we need to prefix these commands as such